### PR TITLE
SPOI-4105 Adding select all functionality to the table

### PR DESCRIPTION
**Pull request by sashadt**

### CODE REVIEW

File: src/directives/mlhrTable.js [hunk 1]
```diff
@@ -215,6 +212,25 @@ angular.module('datatorrent.mlhrTable.directives.mlhrTable', [
...
+
+
+    scope.api = {
+      isSelectedAll: scope.isSelectedAll,
+      selectAll: scope.selectAll,
+      deselectAll: scope.deselectAll,
+      toggleSelectAll: scope.toggleSelectAll,
+
+      setLoading: function(isLoading, triggerDigest) {
+        scope.loading = isLoading;
```
Comment (andyperlitch) [inline src/directives/mlhrTable.js]: this line should actually be:
```js
scope.options.loading = isLoading;
```

**Comment by andyperlitch** [comment]
Looks great. All you have to do now is make that one change I noted above and add some tests to `test/spec/controllers/table-controller.js`.

**Comment by sashadt** [comment]
I updated behavior to select all row contents if selector column is missing, or selector key specified in selector column is undefined in the rows.

**Comment by sashadt** [comment]
Added tests as well.

**Comment by sashadt** [comment]
Updated the dist/* and added error throwing

**Pull request merged**